.pre-commit-config.yaml

```diff
@@ -0,0 +1,49 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.5.0
+  hooks:
+  - id: check-yaml
+  - id: end-of-file-fixer
+  - id: trailing-whitespace
+    exclude: '\.(pdb|gro|top|sdf)$'
+  - id: debug-statements
+- repo: https://github.com/asottile/add-trailing-comma
+  rev: v3.1.0
+  hooks:
+  - id: add-trailing-comma
+- repo: https://github.com/psf/black
+  rev: 24.3.0
+  hooks:
+  - id: black
+  - id: black-jupyter
+- repo: https://github.com/PyCQA/isort
+  rev: 5.13.2
+  hooks:
+  - id: isort
+- repo: https://github.com/PyCQA/flake8
+  rev: 7.0.0
+  hooks:
+  - id: flake8
+- repo: https://github.com/asottile/pyupgrade
+  rev: v3.15.2
+  hooks:
+  - id: pyupgrade
+    args: ["--py39-plus"]
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.3.4
+  hooks:
+    - id: ruff
```

Comment:
With ruff, should be able to drop black, isort, flake8

Reply:
Yep! That's the goal, just took a bad path to get there.

I added it only to get the NumPy 2.0 rules added in, then started adding in some of its other rules, and then got stuck in a huge mess of stuff and put a decided to deal with it later 🫠